php-frontend/src/test/java/org/sonar/php/cfg/ControlFlowGraphTest.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * SonarQube PHP Plugin
+ * Copyright (C) 2010-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.php.cfg;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.sonar.php.PHPTreeModelTest;
+import org.sonar.php.parser.PHPLexicalGrammar;
+import org.sonar.plugins.php.api.tree.declaration.FunctionTree;
+import org.sonar.plugins.php.api.tree.statement.BlockTree;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This CFG Test uses a meta-language to specify the expected structure of the CFG.
+ * See {@link BlockMetadata} for per-basic-block metadata
+ *
+ * Convention:
+ *
+ * 1. the metadata is specified as a method call with the form:
+ * {@code block( [ id => 0, succ => [1,2], pred => [3] ] ); }
+ * where:
+ * - the name of the method is 'block' with one argument
+ * - the argument is a bracketed array with 3 elements:
+ * -- 'id' is the test id of the basic block
+ * -- 'succ' is a bracketed array of expected successor ids
+ * -- 'pred' is a bracketed array of expected predecessor ids
+ *
+ * 2. each basic block must contain the 'block' function call as the first statement
+ *
+ * 3. the metadata string is constructed with the {@link BlockString}
+ */
+public class ControlFlowGraphTest extends PHPTreeModelTest {
```

Comment:
The `PHPTreeModelTest` is an abstract class with 2 methods: `parse` and `expressionToString`

Indeed it should have static methods instead of relying on inheritance, but I should change all the other tests that are using it.